health/health_test.go

```diff
@@ -768,8 +767,19 @@ func TestControlHealthIgnoredOutsideMapPoll(t *testing.T) {
 		t.Error("got a warning with code 'control-health', want none")
 	}
 
-	if gotNotified {
-		t.Error("watcher got called, want it to not be called")
+	// An event is emitted when SetIPNState is run above,
+	// so only fail on the second event.
+	eventCounter := 0
+	expectOne := func(c *Change) error {
```

Comment:
(commentary only)  I have been kinda considering adding a helper to the eventbustest package to do a kind of "negative assertion", a value meaning "either no event at all or an event of some unrelated type".

Reply:
Yeah, the negative one kind of has an implicit timeout requirement ("no event within some reasonable amount of time") that would I think need cooperation from the watcher. Definitely not something we need to stop and do right now, I've just been thinking about this for similar reasons 🙂